spec/cap_spec.rb

```diff
@@ -71,7 +71,11 @@ module Fauxpaas
       let(:source_path) { Pathname.new("some/path/source") }
       let(:shared_path) { Pathname.new("some/path/shared") }
       let(:unshared_path) { Pathname.new("some/path/unshared") }
-      subject { cap.deploy(source_path, shared_path, unshared_path) }
+      let(:artifact) { double(:artifact, source_path: source_path,
```

Comment:
I think we should probably lean toward `instance_double` for more of these, since they verify messages sent would be received. We're not doing any funny mixins or metaprogramming, so I don't think it should cause any/many false alarms.

It sort of moves a little ways into integration testing without actual dependencies on the implementation, which I think is a good thing for validating design changes.

https://relishapp.com/rspec/rspec-mocks/docs/verifying-doubles/using-an-instance-double

Reply:
For future cases, we will use instance doubles where there is a good reason to do so (e.g. large or unstable interface). For now we will leave it as-is here.